src/mbgl/gfx/color_mode.hpp

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include <mbgl/gfx/types.hpp>
+#include <mbgl/util/variant.hpp>
+#include <mbgl/util/color.hpp>
+
+namespace mbgl {
+namespace gfx {
+
+class ColorMode {
+public:
+    template <ColorBlendEquationType E>
+    struct ConstantBlend {
+        static constexpr ColorBlendEquationType equation = E;
+        static constexpr ColorBlendFactorType srcFactor = ColorBlendFactorType::One;
+        static constexpr ColorBlendFactorType dstFactor = ColorBlendFactorType::One;
+    };
+
+    template <ColorBlendEquationType E>
+    struct LinearBlend {
+        static constexpr ColorBlendEquationType equation = E;
+        ColorBlendFactorType srcFactor;
+        ColorBlendFactorType dstFactor;
+    };
+
+    struct Replace {
+        static constexpr ColorBlendEquationType equation = ColorBlendEquationType::Add;
+        static constexpr ColorBlendFactorType srcFactor = ColorBlendFactorType::One;
+        static constexpr ColorBlendFactorType dstFactor = ColorBlendFactorType::Zero;
+    };
+
+    using Add              = LinearBlend<ColorBlendEquationType::Add>;
+    using Subtract         = LinearBlend<ColorBlendEquationType::Subtract>;
+    using ReverseSubtract  = LinearBlend<ColorBlendEquationType::ReverseSubtract>;
+
+    using BlendFunction = variant<
+        Replace,
+        Add,
+        Subtract,
+        ReverseSubtract>;
+
+    BlendFunction blendFunction;
+    Color blendColor;
+
+    struct Mask {
+        bool r;
+        bool g;
+        bool b;
+        bool a;
+    };
+
+    Mask mask;
+
+    static ColorMode disabled() {
```

Comment:
here and below (in other structs), could we make the "factory" methods `constexpr` too?